File: backend/src/modules/inventory/inventory.controller.ts
import { Controller, Get, Param, Query, UseGuards } from '@nestjs/common'
import { JwtAuthGuard } from '../auth/guards/jwt-auth.guard'
import { InventoryService, InventoryFilter } from './inventory.service'

@UseGuards(JwtAuthGuard)
@Controller('inventory')
export class InventoryController {
  constructor(private readonly service: InventoryService) {}

  @Get()
  findAll(@Query() filter: InventoryFilter) {
    return this.service.findAll(filter)
  }

  @Get('low-stock')
  getLowStock(@Query('warehouseId') warehouseId: string) {
    return this.service.getLowStock(warehouseId)
  }

  @Get('summary')
  getSummary(@Query('warehouseId') warehouseId: string) {
    return this.service.getSummary(warehouseId)
  }

  @Get('by-product/:productId')
  findByProduct(
    @Param('productId') productId: string,
    @Query('warehouseId') warehouseId?: string,
  ) {
    return this.service.findByProduct(productId, warehouseId)
  }
}

File: backend/src/main/java/com/wmspro/domain/inventory/InventoryController.java
package com.wmspro.domain.inventory;

import com.wmspro.common.ApiResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;
import java.util.UUID;

@RestController
@RequestMapping("/api/v1/inventory")
@RequiredArgsConstructor
public class InventoryController {

    private final InventoryService inventoryService;

    @GetMapping
    public ApiResponse<List<Inventory>> findAll(@RequestParam UUID warehouseId) {
        return ApiResponse.ok(inventoryService.findAllByWarehouse(warehouseId));
    }

    @GetMapping("/by-product/{productId}")
    public ApiResponse<List<Inventory>> findByProduct(
        @PathVariable UUID productId,
        @RequestParam(required = false) UUID warehouseId
    ) {
        return ApiResponse.ok(inventoryService.findByProduct(productId, warehouseId));
    }

    @GetMapping("/low-stock")
    public ApiResponse<List<Map<String, Object>>> getLowStock(@RequestParam UUID warehouseId) {
        return ApiResponse.ok(inventoryService.findLowStock(warehouseId));
    }

    @GetMapping("/summary")
    public ApiResponse<Map<String, Object>> getSummary(@RequestParam UUID warehouseId) {
        return ApiResponse.ok(inventoryService.getSummary(warehouseId));
    }
}
